src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs

```diff
@@ -9,20 +9,46 @@ namespace NuGet.Configuration
 {
     public class PackageNamespacesConfiguration
     {
+        /// <summary>
+        /// Max allowed length for package Id.
+        /// In case update this value please update in src\NuGet.Core\NuGet.Packaging\PackageCreation\Utility\PackageIdValidator.cs too.
+        /// </summary>
+        internal static int PackageIdMaxLength { get; } = 100;
+
         /// <summary>
         /// Source name to package namespace list.
         /// </summary>
-        public Dictionary<string, IReadOnlyList<string>> Namespaces { get; }
+        internal Dictionary<string, IReadOnlyList<string>> Namespaces { get; }
+
+        private Lazy<SearchTree> SearchTree { get; }
+
+        /// <summary>
+        /// Indicate if any packageSource exist in package namespace section
+        /// </summary>
+        public bool AreNamespacesEnabled { get; }
+
+        /// <summary>
+        /// Get package source names with matching prefix "term" from package namespaces section.
+        /// </summary>
+        /// <param name="term">Search term. Cannot be null, empty, or whitespace only. </param>
+        /// <returns>Package source names with matching prefix "term" from package namespaces.</returns>
+        /// <exception cref="ArgumentException"> if <paramref name="term"/> is null, empty, or whitespace only.</exception>
+        public IReadOnlyList<string> GetConfiguredPackageSources(string term)
+        {
+            return SearchTree.Value?.GetConfiguredPackageSources(term);
```

Comment:
nit: the value can't be null can it?

Reply:
It can be null. If `AreNamespacesEnabled` is false then we get `null` for SearchTree.
https://github.com/NuGet/NuGet.Client/blob/7d97b25df79dbed2923dad02b1842a221535fe70/src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs#L75-L82